homewok_20/PetrolStation.java

```diff
@@ -0,0 +1,38 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class PetrolStation {
+    private int amount;
+    private final Lock lock = new ReentrantLock();
```

Comment:
A lock doesn't solve the problem, that you can refuel no more than 3 threads at once. Try using some synchronizer like a semaphore or any other of your choice.